packages/backend/code/index.ts

```diff
@@ -107,11 +123,79 @@ io.on("connection", (socket) => {
   };
   allPlayers.push(newPlayer);
   // Add server listener for revealCell
-  socket.on("revealCell", revealCell);
-  socket.on("becomeSpymaster", becomeSpymaster);
+  socket.on("revealCell", (cellIndex: number, code: string) => {
+    const color = rooms[code].masterBoard[cellIndex].color;
```

Comment:
Adding this typecast should fix the build. `color` is optional on the cell so it could be `undefined`, but since this is the masterBoard we can be confident it will not be undefined and just typecast it.
```suggestion
    const color = rooms[code].masterBoard[cellIndex].color as Color;
```

Reply:
If we want to be fancy later on, we could add something like a `MasterBoardData` that has cell colors as required.

Reply:
Oh yea, I forgot that was stumping me a bit but I knew it was probably not too big a deal. Figured it had to do with the CellData color being optionally undefined. Still need to clean up a lot then we can work out the merge conflicts.